### PR TITLE
syslinux: 2015-11-09 -> 2019-02-07

### PR DESCRIPTION
**Pull request by lblasc**
###### Motivation for this change
Bump syslinux to 2019-02-06 following debian packaging:
* gPXE is removed in upstream
* added few meaningful patches from debian
* switched to python 3

syslinux is blocker for #66528

Thanks!
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by samueldr** [comment]
Hi, thanks for the PR!

I don't know if you had seen #55500, which had to be reverted? I haven't followed through and re-updated, so that's a great help.

Though, noted there:

> ```
>  # This is syslinux-6.04-pre3^1; syslinux-6.04-pre3 fails to run.
>  # Same issue here https://www.syslinux.org/archives/2019-February/026330.html
> ```

I had to go with b40487005223a78c3bb4c300ef6c436b3f6ec1f7, as the next commit didn't boot once built.

```diff
diff --git a/pkgs/os-specific/linux/syslinux/default.nix b/pkgs/os-specific/linux/syslinux/default.nix
index 0e0f4c945b6..e36a9274b65 100644
--- a/pkgs/os-specific/linux/syslinux/default.nix
+++ b/pkgs/os-specific/linux/syslinux/default.nix
@@ -3,10 +3,12 @@
 stdenv.mkDerivation {
   name = "syslinux-2019-02-06";
 
+  # This is syslinux-6.04-pre3^1; syslinux-6.04-pre3 fails to run.
+  # Same issue here https://www.syslinux.org/archives/2019-February/026330.html
   src = fetchFromRepoOrCz {
     repo = "syslinux";
-    rev = "05ac953c23f90b2328d393f7eecde96e41aed067";
-    sha256 = "07a9jfm5957daziq28vdnb21gnlfsvc85aqyf3fz8hxbpx1krl1l";
+    rev = "b40487005223a78c3bb4c300ef6c436b3f6ec1f7";
+    sha256 = "1qrxl1114sr2i2791z9rf8v53g200aq30f08808d7i8qnmgvxl2w";
   };
 
   patches = let
```

This can be tested using:

```
nix-build nixos/tests/boot.nix -A biosCdrom
```

Which ends hanging at:

```
machine # Booting from DVD/CD...
machine #
machine # ISOLINUX 6.04  ETCD Copyright (C) 1994-2015 H. Peter Anvin et al
machine #
machine # Failed to load ldlinux.c32
machine # Boot failed: press a key to retry...
```

**Review by samueldr — DISMISSED** [review]
(See other comment.)

File: pkgs/os-specific/linux/syslinux/default.nix [hunk 1]
```diff
@@ -1,26 +1,26 @@
-{ stdenv, fetchFromGitHub, fetchurl, nasm, perl, python, libuuid, mtools, makeWrapper }:
+{ stdenv, fetchFromRepoOrCz, fetchurl, nasm, perl, python3, libuuid, mtools, makeWrapper }:
 
 stdenv.mkDerivation {
-  name = "syslinux-2015-11-09";
+  name = "syslinux-2019-02-06";
```
Comment (samueldr) [inline pkgs/os-specific/linux/syslinux/default.nix]: `05ac953c23f90b2328d393f7eecde96e41aed067` is `2019-02-20`

**Comment by lblasc** [comment]
I missed #55500. With `b40487005223a78c3bb4c300ef6c436b3f6ec1f7` and debian patches:
```nix-build nixos/tests/boot.nix -A biosCdrom```
works just fine.

I think this is good to go for `staging`.

**Comment by samueldr** [comment]
Question: why staging? This is has a really limited rebuild impact. Additionally it doesn't have a wide-ranging impact that needs to be tested with other packages, mainly bootloader tests and the isos.

Am I missing something?

**Comment by lblasc** [comment]
@samueldr I was under impression this would yield mass rebuilds (most things I change those days are like that :) ).

Did a rebase on latest master.

**Comment by samueldr** [comment]
@GrahamcOfBorg eval

Something's weird with the eval. Likely the branch switch-a-roo.

**Review by samueldr — CHANGES_REQUESTED** [review]
Oh, looks like your fixes as previously requested were lost. Compare with [the previous force push](https://github.com/NixOS/nixpkgs/commit/a2da91a90f2bbb25b891613b854cbfff4ca83f19). It's currently like the initial PR state.

 * Version number has reverted.
 * Revision has reverted.

Likely trivial to fix :).

**Comment by lblasc** [comment]
I accidentally pushed the old branch, now it should be ok.

**Review by samueldr — APPROVED** [review]

**Comment by samueldr** [comment]
Made a [small whitespace correction](https://github.com/NixOS/nixpkgs/compare/b6bdefb8cd6c73ebb2e35d7b47fe8fb80486d59c..13555cb1ceb50c14841b1daf4e4a92b1eca4f5fb). Didn't have the heart to "reject" the PR for only a byte :).

**Comment by samueldr** [comment]
@vcunat I still don't know what happened the last time we bumped syslinux that needed it to be reverted. If things break again, this PR updated syslinux.

**Comment by vcunat** [comment]
Last time `syslinux` itself failed to build.  Now it does, so hopefully it will hold :-)

**Pull request merged**